### PR TITLE
SOC-3248 - Don't request anon discussions profiles

### PR DESCRIPTION
**Pull request by katanka**
Anonymous users should not trigger a request for Discussion post counts

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 43 issues
* MAJOR 15 major
* MINOR 28 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [UserProfilePageController.class.php](https://github.com/Wikia/app/blob/2c2a3146053dbaaf3ddf93f13118aca2b6b10523/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php): File "UserProfilePageController.class.php" has 1093 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS104)
1. MAJOR [UserProfilePageController.class.php#L6](https://github.com/Wikia/app/blob/2c2a3146053dbaaf3ddf93f13118aca2b6b10523/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L6): Class "UserProfilePageController" has 30 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. MAJOR [UserProfilePageController.class.php#L67](https://github.com/Wikia/app/blob/2c2a3146053dbaaf3ddf93f13118aca2b6b10523/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L67): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. MAJOR [UserProfilePageController.class.php#L167](https://github.com/Wikia/app/blob/2c2a3146053dbaaf3ddf93f13118aca2b6b10523/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L167): The Cyclomatic Complexity of this function "renderActionButton" is 29 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. MAJOR [UserProfilePageController.class.php#L420](https://github.com/Wikia/app/blob/2c2a3146053dbaaf3ddf93f13118aca2b6b10523/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L420): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MAJOR [UserProfilePageController.class.php#L695](https://github.com/Wikia/app/blob/2c2a3146053dbaaf3ddf93f13118aca2b6b10523/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L695): Reduce the number of returns of this function 6, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MAJOR [UserProfilePageController.class.php#L698](https://github.com/Wikia/app/blob/2c2a3146053dbaaf3ddf93f13118aca2b6b10523/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L698): Remove the code after this "return". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1763)
1. MAJOR [UserProfilePageController.class.php#L702](https://github.com/Wikia/app/blob/2c2a3146053dbaaf3ddf93f13118aca2b6b10523/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L702): Remove the code after this "return". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1763)
1. MAJOR [UserProfilePageController.class.php#L706](https://github.com/Wikia/app/blob/2c2a3146053dbaaf3ddf93f13118aca2b6b10523/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L706): Remove the code after this "return". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1763)
1. MAJOR [UserProfilePageController.class.php#L709](https://github.com/Wikia/app/blob/2c2a3146053dbaaf3ddf93f13118aca2b6b10523/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L709): Remove the code after this "return". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1763)

File: extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php [hunk 1]
```diff
@@ -970,7 +970,7 @@ public function fetchDiscussionPostsNumberFrom( $user ) {
 		global $wgEnableDiscussions;
 
 		$this->setVal( 'discussionPostsCountInUserIdentityBoxEnabled', $wgEnableDiscussions );
-		if ( $wgEnableDiscussions ) {
+		if ( $wgEnableDiscussions && $user->isLoggedIn() ) {
```
Comment (armonr) [inline extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php]: Although correct, this method is a bit odd looking in this context considering this is not the session user. For readability's sake, I prefer `!$user->isAnon()` (which calls `isLoggedIn` 😸 )

**Comment by armonr** [comment]
One minor comment, but LGTM

File: extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php (hunk 1 above)
Comment (jsutterfield) [reply extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php]: I agree, that is kind of confusing. `!$user->isAnon()` sounds good to me, or even just `$user->getId() != 0`. Also, maybe we could update that var name to `$targetUser`? That would also make it more clear that we're not referencing the session user.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 43 issues
* MAJOR 15 major
* MINOR 28 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [UserProfilePageController.class.php](https://github.com/Wikia/app/blob/62eb522ed9d8306b07bc73169e53566b8fef0f0a/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php): File "UserProfilePageController.class.php" has 1093 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS104)
1. MAJOR [UserProfilePageController.class.php#L6](https://github.com/Wikia/app/blob/62eb522ed9d8306b07bc73169e53566b8fef0f0a/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L6): Class "UserProfilePageController" has 30 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. MAJOR [UserProfilePageController.class.php#L67](https://github.com/Wikia/app/blob/62eb522ed9d8306b07bc73169e53566b8fef0f0a/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L67): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. MAJOR [UserProfilePageController.class.php#L167](https://github.com/Wikia/app/blob/62eb522ed9d8306b07bc73169e53566b8fef0f0a/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L167): The Cyclomatic Complexity of this function "renderActionButton" is 29 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. MAJOR [UserProfilePageController.class.php#L420](https://github.com/Wikia/app/blob/62eb522ed9d8306b07bc73169e53566b8fef0f0a/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L420): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MAJOR [UserProfilePageController.class.php#L695](https://github.com/Wikia/app/blob/62eb522ed9d8306b07bc73169e53566b8fef0f0a/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L695): Reduce the number of returns of this function 6, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MAJOR [UserProfilePageController.class.php#L698](https://github.com/Wikia/app/blob/62eb522ed9d8306b07bc73169e53566b8fef0f0a/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L698): Remove the code after this "return". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1763)
1. MAJOR [UserProfilePageController.class.php#L702](https://github.com/Wikia/app/blob/62eb522ed9d8306b07bc73169e53566b8fef0f0a/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L702): Remove the code after this "return". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1763)
1. MAJOR [UserProfilePageController.class.php#L706](https://github.com/Wikia/app/blob/62eb522ed9d8306b07bc73169e53566b8fef0f0a/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L706): Remove the code after this "return". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1763)
1. MAJOR [UserProfilePageController.class.php#L709](https://github.com/Wikia/app/blob/62eb522ed9d8306b07bc73169e53566b8fef0f0a/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L709): Remove the code after this "return". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1763)

**Review by armonr — APPROVED** [review]
LGTM!

**Comment by WikiaTech** [comment]
SonarQube analysis reported 43 issues
* MAJOR 15 major
* MINOR 28 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [UserProfilePageController.class.php](https://github.com/Wikia/app/blob/61912884479f59619667b045278a91f440b6dc43/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php): File "UserProfilePageController.class.php" has 1097 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS104)
1. MAJOR [UserProfilePageController.class.php#L6](https://github.com/Wikia/app/blob/61912884479f59619667b045278a91f440b6dc43/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L6): Class "UserProfilePageController" has 30 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. MAJOR [UserProfilePageController.class.php#L67](https://github.com/Wikia/app/blob/61912884479f59619667b045278a91f440b6dc43/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L67): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. MAJOR [UserProfilePageController.class.php#L167](https://github.com/Wikia/app/blob/61912884479f59619667b045278a91f440b6dc43/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L167): The Cyclomatic Complexity of this function "renderActionButton" is 29 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. MAJOR [UserProfilePageController.class.php#L420](https://github.com/Wikia/app/blob/61912884479f59619667b045278a91f440b6dc43/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L420): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MAJOR [UserProfilePageController.class.php#L695](https://github.com/Wikia/app/blob/61912884479f59619667b045278a91f440b6dc43/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L695): Reduce the number of returns of this function 6, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. MAJOR [UserProfilePageController.class.php#L698](https://github.com/Wikia/app/blob/61912884479f59619667b045278a91f440b6dc43/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L698): Remove the code after this "return". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1763)
1. MAJOR [UserProfilePageController.class.php#L702](https://github.com/Wikia/app/blob/61912884479f59619667b045278a91f440b6dc43/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L702): Remove the code after this "return". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1763)
1. MAJOR [UserProfilePageController.class.php#L706](https://github.com/Wikia/app/blob/61912884479f59619667b045278a91f440b6dc43/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L706): Remove the code after this "return". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1763)
1. MAJOR [UserProfilePageController.class.php#L709](https://github.com/Wikia/app/blob/61912884479f59619667b045278a91f440b6dc43/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L709): Remove the code after this "return". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1763)

**Pull request merged**